powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/entity/OperationEntity.java

```diff
@@ -0,0 +1,391 @@
+     * Set operation status.
+     * @param status Operation status.
+     */
+    public void setStatus(OperationStatusDo status) {
+        this.status = status;
+    }
+
+    /**
+     * Get signature type.
+     * @return Signature type.
+     */
+    public PowerAuthSignatureTypes[] getSignatureType() {
+        return signatureType;
+    }
+
+    /**
+     * Set signature type.
+     * @param signatureType Signature type.
+     */
+    public void setSignatureType(PowerAuthSignatureTypes[] signatureType) {
+        this.signatureType = signatureType;
+    }
+
+    /**
+     * Get failure count.
+     * @return Failure count.
+     */
+    public Long getFailureCount() {
+        return failureCount;
+    }
+
+    /**
+     * Set failure count.
+     * @param failureCount Failure count.
+     */
+    public void setFailureCount(Long failureCount) {
+        this.failureCount = failureCount;
+    }
+
+    /**
+     * Get maximum allowed failure count.
+     * @return Maximum allowed failure count.
+     */
+    public Long getMaxFailureCount() {
+        return maxFailureCount;
+    }
+
+    /**
+     * Set maximum allowed failure count.
+     * @param maxFailureCount Maximum allowed failure count.
+     */
+    public void setMaxFailureCount(Long maxFailureCount) {
+        this.maxFailureCount = maxFailureCount;
+    }
+
+    /**
+     * Get timestamp created.
+     * @return Timestamp created.
+     */
+    public Date getTimestampCreated() {
+        return timestampCreated;
+    }
+
+    /**
+     * Set timestamp created.
+     * @param timestampCreated Timestamp created.
+     */
+    public void setTimestampCreated(Date timestampCreated) {
+        this.timestampCreated = timestampCreated;
+    }
+
+    /**
+     * Get timestamp when operation expired.
+     * @return Timestamp when operation expires.
+     */
+    public Date getTimestampExpires() {
+        return timestampExpires;
+    }
+
+    /**
+     * Get timestamp when operation expired.
+     * @param timestampExpires Timestamp when operation expires.
+     */
+    public void setTimestampExpires(Date timestampExpires) {
+        this.timestampExpires = timestampExpires;
+    }
+
+    /**
+     * Get timestamp in which the operation was finalized (moved from PENDING state).
+     * @return Timestamp when operation was finalized.
+     */
+    public Date getTimestampFinalized() {
+        return timestampFinalized;
+    }
+
+    /**
+     * Set timestamp in which the operation was finalized (moved from PENDING state).
+     * @param timestampFinalized Timestamp when operation was finalized.
+     */
+    public void setTimestampFinalized(Date timestampFinalized) {
+        this.timestampFinalized = timestampFinalized;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof OperationEntity)) return false;
+        OperationEntity that = (OperationEntity) o;
+        return Objects.equals(id, that.id)
```

Comment:
I am not sure about this implementation of equals. For instance by increasing the failure counter the object becomes a different object. I think it is safer to use equals with fields which define the object and not use fields which are its attributes which change frequently.